### ISSUE TITLE
`cvmfs_server check -c` intermittently failing on cms.cern.ch

### ISSUE THREAD

**Issue by vvolkl** [cvmfs/cvmfs#3703]
Received a report of the check on cms.cern.ch failing with:
```
[inspecting catalog] ab7f18a18fad2f86ddcc68d62e26cb7528fa139f at /slc7_amd64_gcc630/cms/cmssw/CMSSW_10_1_0_pre3
cvmfs_swissknife: /home/sftnight/jenkins/workspace/CvmfsFullBuildDocker/CVMFS_BUILD_ARCH/docker-x86_64/CVMFS_BUILD_PLATFORM/cc9/build/BUILD/cvmfs-2.11.2/cvmfs/swissknife_check.cc:719: catalog::Catalog* swissknife::CommandCheck::FetchCatalog(const string&, const shash::Any&, uint64_t): Assertion `catalog_file_size > 0' failed.
Aborted (core dumped)

real    17m13.294s
user    6m15.773s
sys     0m59.049s
```
This fails fairly consistently, but not always and not always at the same spot. Additional logging will be helpful for further debugging

**Comment by vvolkl** [comment]
This has been brought up again, and I think this is due to `cvmfs_server check` using `/var/spool/cvmfs/<repo>/tmp` as a scratch directory. In particular other commands like `cvmfs_server abort` may clear out this directory, and if this happens while check has downloaded a file, but not checked it yet, it triggers the above error. #4011 should let us confirm if that is the issue, and if it is, the default for the temp dir  should be changed.

**Comment by ffurano** [comment]
Hi Valentin, so are you suggesting to add this option whenever "check -c" is run ?

**Comment by vvolkl** [comment]
> Hi Valentin, so are you suggesting to add this option whenever "check -c" is run ?

Yes, if you see this error you should add the option. The default should be changed eventually, but better to do it in a minor release than in a patch release.

**Comment by DrDaveD** [comment]
Does this problem only happen on release manager/publisher machines?  Because on stratum 1s we don't have transactions to abort, and we only clean out the txn directories of files that are couple of days old.

**Comment by vvolkl** [comment]
> Does this problem only happen on release manager/publisher machines? Because on stratum 1s we don't have transactions to abort, and we only clean out the txn directories of files that are couple of days old.

If I understand the problem correctly, it'd mostly happen on stratum-0s, but it could theoretically also happen on a stratum-1 that the mountpoint is reloaded. @ffurano may comment if he saw this on stratum-1s also.

**Comment by DrDaveD** [comment]
> it could theoretically also happen on a stratum-1 that the mountpoint is reloaded

What mountpoint?  Stratum 1s don't mount anything.
